.github/workflows/pull-request.yml

```diff
@@ -47,3 +47,50 @@ jobs:
           packagePAT: ${{ secrets.GITHUB_TOKEN }}
           JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF8
         run: ./gradlew.bat build --stacktrace --scan --console=plain --no-daemon
+
+  sonarcloud_scan:
+    name: Build with sonarcloud scan on Ubuntu
+    runs-on: ubuntu-latest
+    timeout-minutes: 120
+    concurrency:
+      group: ${{ github.head_ref }}-ubuntu
+      cancel-in-progress: true
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: Set up JDK 11
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+
+      - name: Initialize sub-modules
+        run: git submodule update --init
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Build with Gradle
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
pass env variables as in https://github.com/ballerina-platform/openapi-tools/blob/master/.github/workflows/pull-request.yml#L25

Reply:
add both packageUser and packagePAT

Reply:
Updated